packages/governance/src/paramGovernance/typedParamManager.js

```diff
@@ -0,0 +1,105 @@
+// @ts-check
+
+import { makeParamManagerBuilder } from './paramManager.js';
+
+/**
+ * @template {Record<Keyword, ParamRecordTuple>} T
+ * @typedef {{
+ *   [Property in keyof T as `get${string & Property}`]: () => ParamValueForType<T[Property][0]>
+ * }} Getters
+ */
+
+/**
+ * @template {Record<Keyword, ParamRecordTuple>} T
+ * @typedef {{
+ *   [Property in keyof T as `update${string & Property}`]: (value: ParamValueForType<T[Property][0]>) => void
+ * }} Updaters
+ */
+
+/**
+ * @template {Record<Keyword, ST<ParamType>>} T
+ * @typedef {ParamManagerBase & Getters<T> & Updaters<T> & {asGetters: () => Getters<T>}} TypedParamManager
+ */
+
+/**
+ * @param {ParamType} type
+ */
+const builderMethodName = type =>
+  `add${type[0].toUpperCase() + type.substring(1)}`;
```

Comment:
I worry that allowing callers to use names that start with lower case and converting will run into some of the issues that drove us to insist Zoe keywords have initial capital letters. I don't understand those issues well enough to say whether we're immune to them.

I concede that it's more friendly to correct mistakes than to turn them into errors. I think the cost of the errors is only at initial contract writing time, and not an ongoing burden.

Reply:
In this case we have a guarantee of the domain of `type[0]` and thus all the strings. If somehow a bad string got in, it would just call a different `add` method on the builder.